wasm/__init__.py

```diff
@@ -4602,45 +4599,62 @@ def type_table(store, tableaddr):
 
 def read_table(store, tableaddr, i):
     if len(store["tables"]) < tableaddr:
-        return "error"
+        raise ValidationError(
+            f"Table address outside of allowed range: {tableaddr} > "
+            f"{len(store['tables'])}"
+        )
```

Comment:
Lots of repetitive validation code here but I'm inclined to clean this up later rather than trying to DRY it now since I see a pretty clear path to embedding this logic in a handful of `Globals/Tables/Functions/Memories` objects with well defined external APIs that wrap the validation.